File: C++_Base_PT4/XText56.cpp
#include "pt4.h"
#include <fstream>
using namespace std;

void Solve()
{
    Task("XText56");
	string s1, s2;
	
	pt >> s1 >> s2;
	
	fstream f1, f2;
	
	f1.open(s1, ios::in);
	
	f2.open(s2, ios::binary | ios::out);
	
	int i = 0;
	
	char ch[100];
	
	while(f1.peek() != -1)
	{
		char x;
		f1 >> x;
//		ShowLine(x);
		int k = 0;
		for(int j = 0; j < i; j ++)
		{
			if(x == ch[j])
			{
				k = 1;
				break;
			}
		}
		if(k == 0)
		{
			ch[i] = x;
//			ShowLine(ch[i]);
			i++;
		}
	}
	
	for(int m = 0; m < i; m++)
	{
		for(int n = m + 1; n < i; n++)
		{
			if(ch[m] < ch[n])
			{
				char r;
				r = ch[m];
				ch[m] = ch[n];
				ch[n] = r;
			}
		}
	}
	
	for(int k = 0; k < i; k++)
	{
		char c = ch[k];
		Show(c);
		f2.write((char *)&c, sizeof(c)); 
	}
	char c = ' ';
	Show(c);
	f2.write((char *)&c, sizeof(c)); 

	f1.close();
	f2.close();
}

File: C#_Base_PT4/XText56.cs
// File: "XText56"
using PT4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static void Solve()
        {
            Task("XText56");

            string s1 = GetString();
            string s2 = GetString();

            FileStream f1 = new FileStream(s1, FileMode.Open);
            FileStream f2 = new FileStream(s2, FileMode.Create);

            BinaryWriter w = new BinaryWriter (f2);
            BinaryReader r = new BinaryReader (f1);

            int i = 0;
            char[] ch = new char [100];
            while(f1.Position!=f1.Length)
            {
                char x = (char)r.Read();
                Show(x);
                int k = 0;
                for(int j = 0; j< i; j++)
                {
                    if(x == ch[j])
                    {
                        k = 1;
                        break;
                    }
                }
                if(k == 0)
                {
                    ch[i] = x;
                    i++;
                }
            }
            char q;
            for(int m = 0; m < i; m++)
            {
                for(int n = m + 1; n < i; n++)
                {
                    if(ch[m] < ch[n])
                    {
                        q = ch[m];
                        ch[m] = ch[n];
                        ch[n] = q;
                    }
                }
            }

            for(int k = 0; k < i-3; k++)
            {
                char c = ch[k];
                Show(c);
                w.Write(c); 
            }

            char p = ' ';
	        Show(p);
	        w.Write(p); 

            r.Close();
            w.Close();
        }
    }
}
